File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    answer = 0
    cache = []
    for i in cities:
        i = i.upper()
        if i in cache:
            cache.remove(i)
            cache.insert(0, i)
            cache = cache[:cacheSize]
            answer+=1
        else:
            cache.insert(0, i)
            cache = cache[:cacheSize]
            answer+=5
    return answer

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.js
function solution(cacheSize, cities) {
    const cache = []
    let runtime = 0
    
    if (cacheSize === 0) return cities.length * 5
    
    for(city of cities) {
        const upperCity = city.toUpperCase()
        
        if(cache.includes(upperCity)) {
            const idx = cache.findIndex((val) => val === upperCity)
            cache.splice(idx, 1)
            cache.push(upperCity)
            runtime+=1
        } else {
            if(cache.length === cacheSize) cache.shift()
            cache.push(upperCity)
            runtime+=5
        }
    }
    
    return runtime
}